### PR TITLE
[RDY] Cargo pants cutoffs

### PR DESCRIPTION
**Pull request by railmonkey**
Adds a quick and dirty cargo shorts recipe to be in line with the rest of the cutoff recipes. Model was denim shorts.

### CODE REVIEW

File: data/json/recipes/armor/legs.json [hunk 1]
```diff
@@ -450,6 +451,18 @@
...
   },
   {
+    "result": "shorts_cargo",
+    "type": "recipe",
+    "category": "CC_ARMOR",
+    "subcategory": "CSC_ARMOR_LEGS",
+    "skill_used": "tailor",
+    "time": 30000,
+    "autolearn": true,
+    "qualities": [ { "id": "CUT", "level": 1, "amount": 1 } ],
```
Comment (Coolthulhu) [inline data/json/recipes/armor/legs.json]: "`"amount": 1`" is redundant and should be removed.

**Comment by pisskop** [comment]
This provides an easy way to clean dirty clothing.

But then again cargo shorts are like tailoring 2

**Comment by railmonkey** [comment]
Huh. Forgot about Squeamish. Should the rest of the cutoff recipes be reviewed, then?

And yeah, that's my main motivation. For whatever reason, cargo shorts were overlooked. Anyone can take a sharp thing to a pair of pants, as long as they aren't concerned about looks. It's sewing new hems that needs sewing knowledge, and not much of that.

**Comment by Coolthulhu** [comment]
Squeamish is inherently broken without flag passing. It shouldn't be treated too seriously.

**Comment by railmonkey** [comment]
In that case, this should be ready for merging.

File: data/json/recipes/armor/legs.json [hunk 2]
```diff
@@ -439,6 +439,7 @@
   {
     "result": "shorts_cargo",
     "type": "recipe",
+    "id_suffix": "from fabric",
```
Comment (Coolthulhu) [inline data/json/recipes/armor/legs.json]: Do other values of "id_suffix" have spaces in them?

Comment (mugling) [inline data/json/recipes/armor/legs.json]: It's not forbidden but it's very inconsistent

Comment (railmonkey) [inline data/json/recipes/armor/legs.json]: All but an anomalous "from_stockings". Should I underscore the rest or just get rid of that single underscore?

File: data/json/recipes/armor/legs.json (hunk 1 above)
Comment (mugling) [inline data/json/recipes/armor/legs.json]: Avoid making unrelated change to `amount: 1` elsewhere though

File: data/json/recipes/armor/legs.json [hunk 3]
```diff
@@ -439,6 +439,7 @@
   {
     "result": "shorts_cargo",
     "type": "recipe",
+    "id_suffix": "from_fabric",
```
Comment (mugling) [inline data/json/recipes/armor/legs.json]: The version from rags should be the default (no suffix). The one using scissors to cut off the bottom should be an alternative (with a suffix)

Comment (railmonkey) [inline data/json/recipes/armor/legs.json]: Ah, okay. Life happened, but I'll fix that now.

File: data/json/recipes/armor/legs.json [hunk 4]
```diff
@@ -450,9 +450,22 @@
     "components": [ [ [ "rag", 6 ] ] ]
   },
   {
+    "result": "shorts_cargo",
+    "type": "recipe",
+    "id_suffix": "from_pants"
```
Comment (Cyrano7) [inline data/json/recipes/armor/legs.json]: You still there? All you need is one comma at the end of this line and its gtg.

**Comment by BorkBorkGoesTheCode** [comment]
@railmonkey ping
